File: api-node-pixel/src/app/models/dadoClimatico.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const dadoClimaticoSchema = new Schema({
  tipo: { type: String, required: true }, // temperatura, umidade, chuva
  valor: { type: Number, required: true },
  dataHora: { type: Date, required: true },
  latitude: { type: Number, required: false },
  longitude: { type: Number, required: false },
  estacao: { type: Schema.Types.ObjectId, ref: 'Estacao', required: true }
});

module.exports = mongoose.model('DadoClimatico', dadoClimaticoSchema);

File: Pixel/src/main/java/br/edu/utfpr/api/model/DadoClimatico.java
package br.edu.utfpr.api.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "tb_dado_climatico")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class DadoClimatico extends BaseEntity {

    @Column(nullable = false)
    private String tipo; // temperatura, umidade, chuva

    @Column(nullable = false)
    private double valor;

    @Column(nullable = false)
    private LocalDateTime dataHora;

    @Column(nullable = true)
    private Double latitude;

    @Column(nullable = true)
    private Double longitude;

    @ManyToOne
    @JoinColumn(name = "estacao_id")
    private Estacao estacao;
}
